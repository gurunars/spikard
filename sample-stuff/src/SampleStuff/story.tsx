import { action } from "@storybook/addon-actions"
import { storiesOf } from "@storybook/react"
import React from "react"
import { host } from "storybook-host"

import SampleStuff from "../SampleStuff"

storiesOf("SampleStuff", module)
  .addDecorator(host({
    align: "center middle",
    height: 600,
    width: 800,
  }))
  .add("basic", () => (
    <SampleStuff
      title="Sample Title"
      onClick={action("onClick")}
    />
  ))
