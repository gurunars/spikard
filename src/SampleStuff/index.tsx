import React from "react";

import Icon from "./icon.svg";

const SampleStuff = (props: {
  title: string;
  onClick: () => void;
}): React.ReactElement<any> => (
    <div>
      <button
        style={{
          cursor: "pointer",
          alignItems: "center",
          display: "flex",
          padding: "5px",
          color: "green"
        }}
        onClick={props.onClick}
      >
        <Icon color="green" />
        {props.title}
      </button>
    </div>
  );

export default SampleStuff;
