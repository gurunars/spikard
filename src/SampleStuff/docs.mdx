---
name: SampleStuff
route: /
---

import { Playground, Props } from "docz";
import SampleStuff from "./";

# SampleStuff

<Props of={SampleStuff} />

## Basic usage

<Playground>
  <SampleStuff title="Sample Title" onClick={() => console.log("Click")} />
</Playground>
