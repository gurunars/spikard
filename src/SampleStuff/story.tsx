import React from "react";
import { action } from "@storybook/addon-actions";
import { storiesOf } from "@storybook/react";
import { host } from "storybook-host";

import SampleStuff from ".";

storiesOf("SampleStuff", module)
  .addDecorator(host({
    align: "center middle",
    height: 600,
    width: 800,
  }))
  .add("basic", () => (
    <SampleStuff
      title="Sample Title"
      onClick={action("onClick")}
    />
  ));
