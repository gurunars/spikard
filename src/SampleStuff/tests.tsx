import { shallow } from "enzyme";
import React from "react";
import renderer from "react-test-renderer";

import SampleStuff from ".";

describe("SampleStuff", () => {
  it("should render correctly", () => {
    const onClick = jest.fn();
    const tree = renderer
      .create(<SampleStuff title="FooBar" onClick={onClick} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should process clicks correctly", () => {
    const onClick = jest.fn();
    const wrapper = shallow(<SampleStuff title="FooBar" onClick={onClick} />);
    wrapper
      .find("button")
      .at(0)
      .simulate("click");
    expect(onClick).toHaveBeenCalled();
  });
});
