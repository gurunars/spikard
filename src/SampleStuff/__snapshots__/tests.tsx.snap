// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SampleStuff should render correctly 1`] = `
<button
  onClick={[Function]}
  style={
    Object {
      "cursor": "pointer",
    }
  }
>
  FooBar
</button>
`;
